cdk/lib/recipes-backend.ts

```diff
@@ -230,6 +241,11 @@ export class RecipesBackend extends GuStack {
 			dataStore: store,
 		});
 
+		new RecipesReindex(this, 'RecipeReindex', {
+			contentUrlBase,
+			reindexBatchSize: reindexBatchSizeParam.valueAsString,
```

Comment:
Do we need any type and/or availability guards around this?

Reply:
I don't think we can guard against this being a string if we're pulling it from param store — there's no number type there. A few things we can do:

1. Keep this as a stack param, where we can specify a number type. + catch failure at input time (preferable), - require a manual stack update to update (small risk of unintended changes)
2. Better failure mode when this isn't a `number`. + a bit clearer what is happening in logs, - failure on run is much less transparent

Because this is an SSM param, and it is likely to change infrequently, I'm leaning towards 2 — but WDYT?

Reply:
2 sounds good to me, too. Protecting ourselves from it not being a number, or being missing entirely when we try to access it is good 👍

Reply:
This code does only run at build time (it's in the CDK) so a failure won't affect the running app

Reply:
I've gone with 1. for now, as I can't see a way to transform the string param from SSM into a number for the wait fn's `Duration.seconds` constructor (you can go the other way with `stringifyNumber`, but there's no number -> string equivalent AFAICS?) — we can revisit as needed. bc2d9a722e82f55f379430486c51095adeba7b27

Reply:
@jonathonherbert this is because `reindexBatchSizeParam.valueAsString` does _not_ actually return a string. Rather, it gives a "placeholder" which is rendered into the Cloudformation as the equivalent of:

```yaml
reindexBatchSize: !Ref reindexBatchSizeParam
```

This is an advantage because it's evaluated at deploy time not build time.

Therefore there is not way to transform it to a number, because Cloudformation can't support that.

Reply:
I know, it's a token: I suppose there's a way to move from numbers to strings because that is always possible, whereas moving in the other direction can fail 💥